src/Integration.Vsix.UnitTests/ErrorList/SonarErrorListDataSource_IssueLocationStoreTests.cs

```diff
@@ -211,6 +213,62 @@ public void LocationsUpdated_FactoriesWithMatchingFilesUpdatedAndErrorListRefres
             CheckSinkNotifiedOfChangeToFactories(sinkMock2, factoryWithMatch1, factoryWithMatch2);
         }
 
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void LocationsUpdated_SelectedIssueIsNotInFactory_SelectedIssueNotChanged(bool hasSelectedIssue)
+        {
+            var selectedIssue = hasSelectedIssue ? Mock.Of<IAnalysisIssueVisualization>() : null;
+            var selectionService = new Mock<IIssueSelectionService>();
+            selectionService.Setup(x => x.SelectedIssue).Returns(selectedIssue);
+
+            var factory = CreateFactoryWithIssues(Mock.Of<IAnalysisIssueVisualization>());
+            var testSubject = CreateTestSubject(selectionService.Object);
+            testSubject.AddFactory(factory);
+
+            testSubject.Refresh(new[] { "test.txt" });
+
+            selectionService.VerifySet(x => x.SelectedIssue = It.IsAny<IAnalysisIssueVisualization>(), Times.Never);
+        }
+
+        [TestMethod]
+        public void LocationsUpdated_SelectedIssueIsInFactory_IssueIsNavigable_SelectedIssueNotChanged()
+        {
+            var navigableSpan = (SnapshotSpan?) null;
+            var selectedIssue = new Mock<IAnalysisIssueVisualization>();
+            selectedIssue.Setup(x => x.Span).Returns(navigableSpan);
+
+            var selectionService = new Mock<IIssueSelectionService>();
+            selectionService.Setup(x => x.SelectedIssue).Returns(selectedIssue.Object);
+
+            var factory = CreateFactoryWithIssues(selectedIssue.Object);
+            var testSubject = CreateTestSubject(selectionService.Object);
+            testSubject.AddFactory(factory);
+
+            testSubject.Refresh(new[] { "test.txt" });
+
+            selectionService.VerifySet(x => x.SelectedIssue = It.IsAny<IAnalysisIssueVisualization>(), Times.Never);
+        }
+
+        [TestMethod]
+        public void LocationsUpdated_SelectedIssueIsInFactory_IssueIsNotNavigable_SelectedIssueIsCleared()
+        {
+            var nonNavigableSpan = new SnapshotSpan();
+            var selectedIssue = new Mock<IAnalysisIssueVisualization>();
+            selectedIssue.Setup(x => x.Span).Returns(nonNavigableSpan);
+
+            var selectionService = new Mock<IIssueSelectionService>();
+            selectionService.Setup(x => x.SelectedIssue).Returns(selectedIssue.Object);
+
+            var factory = CreateFactoryWithIssues(selectedIssue.Object);
+            var testSubject = CreateTestSubject(selectionService.Object);
+            testSubject.AddFactory(factory);
+
+            testSubject.Refresh(new[] { "test.txt" });
+
+            selectionService.VerifySet(x => x.SelectedIssue = null, Times.Once);
```

Comment:
Minor: the repeated setup boilerplate makes it harder obscures the intent of the test. A helper function would make the intent clearer e.g.

```
var selectionService = CreateSelectionService(selectedIssueViz: null);
...
var selectionService = CreateSelectionService(selectedIssueViz: NaviagableIssueViz);
...
var selectionService = CreateSelectionService(selectedIssueViz: NonNavigableIssueViz);
```